lib/automod/index.js

```diff
@@ -0,0 +1,201 @@
+    ]);
+}
+const doCheckDocs = async () => {
+    for (const doc of currentDocs) {
+        try {
+
+            if(process.env.DEBUG){
+                console.log ('checking', doc.id, doc.data ().id);
+            }
+
+            const meta = doc.data ().meta;
+            const image = getInfuraURL (meta.image);
+            const media = meta.image !== meta.media ? getInfuraURL (meta.media) : getInfuraURL (meta.media);
+            const mime = meta.mimetype || meta.mimeType;
+            // const actualType = await axios.head (media);
+            await setTimeout (1000);
+            if (false) { // (actualType.headers['content-type'] !== mime)
+                // await declineDoc (doc, `wrong mimetype, expected ${mime}, got ${actualType.headers['content-type']}`);
+            } else {
+                try {
+                    if (meta.image !== meta.media && !mime.startsWith ('audio')) {
+                        const resultImagePromise = rekognition.detectExplicitContent ({
+                            url: image,
+                            config: {
+                                resize: { width: 1024 }
+                            }
+                        });
+                        const resultMediaPromise = rekognition.detectExplicitContent ({
+                            url: media
+                        });
+
+                        const [imageResult, mediaResult] = await Promise.all ([resultImagePromise, resultMediaPromise]);
+                        const allLabels = [...imageResult.ModerationLabels, ...mediaResult.ModerationLabels];
+                        const labels = constructLabels (allLabels);
+                        await checkLabels (doc, labels);
+                    } else {
+                        const resultImage = await rekognition.detectExplicitContent ({
+                            url: image,
+                            config: {
+                                resize: { width: 1024 }
+                            }
+                        });
+                        const labels = constructLabels (resultImage.ModerationLabels);
+                        await checkLabels (doc, labels);
+                    }
+                } catch (e) {
+                    if (e.toString ().toLowerCase ().includesAny (
+                        ['video duration', 'invalid file type', 'invalid data content']
+                    )) await declineDoc (doc, e);
+                    else console.error (e);
+                }
+            }
+        } catch (e) {
+            console.error (e);
+            declineDoc (doc, e);
+        }
+        await setTimeout (2000);
+    }
+    currentDocs = []
+    await setTimeout (2000);
+    if (lastUpdated < Date.now () - 1000 * 60 * 3) {
+        await doHealthUpdate ();
+        lastUpdated = Date.now ();
+    }
+    doCheckDocs ();
+}
+
+doCheckDocs ();
+
+let lastUpdated = Date.now ();
+
+
+const initAutomod = ()=>{
```

Comment:
we will deploy separate mainnet/testnet servers so there will be 2 automod instances running - only enable automod in mainnet

Reply:
did it in index.js inside root of project 
`if(defaultNetwork==='reef_mainnet') {
	initAutomod();
}
`